Added cutscene skip for zelda escaping

aMannus opened the pull request:
Using the debug cutscene skipping function. Also added a conditional so the bridge doesn't spawn closed when cutscene is ready to trigger so you don't fall straight into the water if you come from hyrule market when the cutscene is ready to trigger.

The way I'm hijacking the debug cutscene skip _feels_ a bit dirty, especially if we're ever going to use the same method for other cutscene skips. So if there's a better way of doing it, lemme know.

This makes it identical to the N64 cutscene skip. If we want to make it more fancy we can take a look at it later. This is fine for now I'd say.

Looks like this in game: https://cdn.discordapp.com/attachments/988962246585634856/993160744948465664/2022-07-03_16-24-26.mp4

https://github.com/briaguya-ai/rando-issue-tracker/issues/57

aMannus committed:
Added cutscene skip for zelda escaping

Using the debug cutscene skipping function. Also added a conditional so the bridge doesn't spawn closed when cutscene is ready to trigger

briaguya0 commented:
unless i'm reading this wrong it seems like this is trying to skip *every* cutscene when rando'd...

this feels like a really big change that probably has unintended consequences

aMannus commented:
No it only skips the specific cutscene. `if (!(gSaveContext.n64ddFlag) || cmd->base == 33) {` makes sure it's only that one when in rando.

The diff on this is kinda weird because I wrap a large part of the code in another if statement and changed the indentation because of it.

aMannus commented:
OK one second, I wanna change something about this.

briaguya0 commented:
still not fully understanding the logic here, let me know if i'm right

the if block that normally checks for pressing the "skip cutscene" button in debug mode now also lets us in if we're rando'd

before going into the giant switch/case, we check to see if we're rando'd and on a specific scene

i feel like there's gotta be a cleaner way to handle this (i don't think we're doing anything after the switch/case so maybe just early return if we aren't on the right cutscene?)

aMannus committed:
Better implementation for specific rando cutscene skips

aMannus committed:
Merge branch 'pr/416' into zelda-escape-cutscene-skip

aMannus committed:
Cleaned up and fixed zelda escape skip

The if statement is a freaking monster, but unless we want to skip more cutscenes in the same way later, this is the most compact way of doing it that I know of.

aMannus committed:
Revert one line to match original

nothing functional

aMannus commented:
OK so I managed to fix it. Turns out, this function is used for other cutscenes outside of the debug function as well (I didn't fully understand what the if statement was actually checking for), so my old version wouldn't be good anyway.

I managed to just change around the if statement before the massive switch so now the vanilla cutscenes that use this aren't broken either, and should be a lot more clean.

That said, this if statement is an absolute monster, and if we ever do want to skip more cutscenes in rando by using this function, we should probably change this around. I'd also just like some actual developer's eyes on this because maybe there's a much cleaner way I'm unaware of in general.

briaguya0 commented on soh/src/code/z_demo.c:
```diff
@@ -502,8 +502,11 @@ void Cutscene_Command_Terminator(GlobalContext* globalCtx, CutsceneContext* csCt
 
     if ((csCtx->frames == cmd->startFrame) || (temp != 0) ||
+        // Skip specific cutscene in rando automatically
+        (gSaveContext.n64ddFlag && cmd->base == 33 && (csCtx->frames > 20)) ||
```
could you explain what `csCtx->frames > 20` is doing?

what happens if you remove it?

briaguya0 replied:
also probably would be good to mention which cutscene, 33 is kind of a magic number at the moment

briaguya0 reviewed (commented):
looking at lot better, one little question

aMannus committed:
Streamlined OoT cutscene skip for future additions

Also cleaned up if statement in general

aMannus committed:
Made if statement more readable

Also added clarity for what cutscene was skipped

aMannus committed:
Fixed typo in comment
